src/client/builtin_scene/materials/gaussian_splatting.cpp

```diff
@@ -0,0 +1,311 @@
+#include "./gaussian_splatting.hpp"
+#include <client/graphics/webgl_context.hpp>
+#include <client/graphics/webgl_program.hpp>
+#include <client/graphics/webgl_buffer.hpp>
+#include "../meshes.hpp"
+
+namespace builtin_scene::materials
+{
+  bool GaussianSplattingMaterial::initialize(std::shared_ptr<client_graphics::WebGL2Context> glContext,
+                                             std::shared_ptr<client_graphics::WebGLProgram> program)
+  {
+    if (TR_UNLIKELY(!Material::initialize(glContext, program)))
+      return false;
+
+    // Initialize WebGL buffers for instanced gaussian splatting data
+    splatPositionBuffer_ = glContext->createBuffer();
+    splatColorBuffer_ = glContext->createBuffer();
+    splatOpacityBuffer_ = glContext->createBuffer();
+    splatScaleBuffer_ = glContext->createBuffer();
+    splatRotationBuffer_ = glContext->createBuffer();
+
+    // Initialize base quad geometry buffers
+    quadVertexBuffer_ = glContext->createBuffer();
+    quadIndexBuffer_ = glContext->createBuffer();
+
+    // Create the base quad geometry
+    createQuadGeometry(glContext);
+
+    buffersInitialized_ = true;
+    return true;
+  }
+
+  void GaussianSplattingMaterial::createQuadGeometry(std::shared_ptr<client_graphics::WebGL2Context> glContext)
```

Comment:
This can be moved to `GaussianSplatsMesh`

Reply:
And add a new static method `MeshBuilder::CreateGaussianSplatsMesh()` to create gaussian splats mesh.

In summary, there will be two instances-based mesh: InstancesMesh for rendering HTML 2d elements and `SplatsMesh` for rendering HTML splats (in Gaussian Splatting models).